### PR TITLE
Upgrade all packages

### PR DESCRIPTION
**Pull request by adam-e-trepanier**

### CODE REVIEW

**Review by williams4027 — APPROVED** [review]
This looks good!

Typically when updating these dependency projects like root-eslint-plugin, it is nice to get an in-progress PR up that showcases the update on a repo that uses it (in this case, root-mobile). To do that, I will:

- create a branch off of master
- change the package.json to look at this specific commit (`"eslint-plugin-root": "https://github.com/Root-App/root-eslint-plugin#76e22fbb4a46c898f257d628cf6b630e21a4ac55",`)
- run yarn install to make sure I am pulling the dependencies
- run yarn lint locally (all should be good ideally, since no rules were changed)
- push these changes remotely and make sure CI passes

Then I will post that build result here for "proof".

Then once this PR is merged, you can simply go to the root-mobile PR, rebase on master, and update with the master commit SHA to verify all is clear.

**Comment by adam-e-trepanier** [comment]
root-mobile build: https://buildkite.com/root-insurance/root-mobile/builds/9702 with package pointing to this branch : `"eslint-plugin-root": "https://github.com/Root-App/root-eslint-plugin#at-js-upgrade-plugin-packages"`

File: package.json [hunk 1]
```diff
@@ -15,19 +15,19 @@
...
   },
   "dependencies": {
-    "babel-eslint": "^10.0.1",
-    "eslint": "^5.10.0",
-    "eslint-plugin-jest": "^22.1.2",
-    "eslint-plugin-mocha": "^5.2.0",
-    "eslint-plugin-react": "^7.10.0",
-    "eslint-plugin-react-native": "^3.1.0",
+    "babel-eslint": "^10.0.2",
+    "eslint": "^6.0.1",
```
Comment (kylekthompson) [inline package.json:19]: These changes still work with root-web/myles-mobile, right?

Comment (adam-e-trepanier) [reply package.json:19]: Good question.  I am completely unaware of what `root-web/myles-mobile` actually is.  Will have to clone and see if I can spin it up, set the package.json to point to this branch and let CI prove it.   Is there anything else I should be looking at regarding myles?

Comment (kylekthompson) [reply package.json:19]: Nope. It's just linting, so assuming the build passes you're probably ok

**Review by kylekthompson — APPROVED** [review]
👍 pending my comment

File: package.json (hunk 1 above)
Comment (adam-e-trepanier) [reply package.json:19]: myles-mobile passes the minute I remove the rule that no longer exists (convo in slack).

**Comment by adam-e-trepanier** [comment]
@TAGraves responded in slack regarding the await-poll-for-condition rule.  We will be changing it to preceded-by-await.

**Pull request merged**